### PR TITLE
[Module 3]

### PR DESCRIPTION
**Pull request by juasmartinezbelS4N**

### CODE REVIEW

File: module_3/task1.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+myEmitter.emit('eventOne');
+
+// Emit 'eventOnce' -> After this the eventOnce will be
+// removed/unregistered automatically
+myEmitter.emit('eventOnce');
+
+
+myEmitter.emit('eventOne');
+myEmitter.emit('init');
+myEmitter.emit('init'); // Will not be fired
```
Comment (MinMiguelM) [inline module_3/task1.js:33]: it is not being removed after the fist emit, remember init is a once event and should be removed at the first emit, however, it is not. Check the output of this and you will notice the message is being printed twice

File: module_3/task3.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const onError = (err) => {
+      console.error(err)
+    }
+
+    const onComplete = () => {
+      console.log("completed")
+    }
+
+    csvtojson()
+      .fromFile(CSV_FILE_PATH)
```
Comment (MinMiguelM) [inline module_3/task3.js]: using `fromFile` is loading the full file into memory which is not requested in this task, the idea is reading line by line using streamings, check `csvtojson` documentation to see how you could refactor it

File: module_3/task3.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  try {
+    const onError = (err) => {
+      console.error(err)
+    }
+
+    const onComplete = () => {
+      console.log("completed")
+    }
+
+    csvtojson()
```
Comment (MinMiguelM) [inline module_3/task3.js]: the TXT file is containing the `amount` property which is not expected, you should ignore it when writting to TXT file

File: module_3/task3.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          }
+          fs.appendFile(
+            "data-csv.txt",
+            `${JSON.stringify(parsedJson)}\n`,
+            (err) => {
+              if (err) {
+                console.error(err)
+                throw new Error(err)
+              }
+            }
```
Comment (MinMiguelM) [inline module_3/task3.js]: how could you rewrite to avoid callback hell? Small functions?

**Review by MinMiguelM — CHANGES_REQUESTED** [review]
Great job! Almost there

File: module_3/task3.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    const line = byLine[i].substring(1).split(",")
+    line.splice(2, 1)
+    const parsed = {}
+    for (const index in line) {
+      parsed[header[index]] = line[index]
+    }
+    fullItems.push(parsed)
+  }
+  return fullItems
+}
```
Comment (MinMiguelM) [inline module_3/task3.js]: the idea of csvtojson is avoiding you to make this manual work. So check these links and see how you can do it:

* streaming data: https://www.npmjs.com/package/csvtojson#asynchronously-process-each-line-from-csv-url
* IgnoreColumns: https://www.npmjs.com/package/csvtojson#api

**Review by MinMiguelM — CHANGES_REQUESTED** [review]

File: module_3/task3.js (hunk 5 above)
Comment (juasmartinezbelS4N) [reply module_3/task3.js]: I didn't know you could mix subscribe with the pipes. The links don't work, but I get it from the comment itself

Thank you!

**Review by MinMiguelM — APPROVED** [review]
It looks good, thank you for your effort

**Pull request merged**